tools/state-parts-dump/src/rpc_requests.rs

```diff
@@ -0,0 +1,214 @@
+                "id": "dontcare",
+                "method": "validators",
+                "params": {
+                    "epoch_id": epoch_id
+                }
+            }))
+            .send()?
+            .json()?;
+        let epoch_start_height = epoch_response.result.epoch_start_height;
+        let epoch_start_block_hash = self.get_block_hash(epoch_start_height).unwrap();
+        Ok(epoch_start_block_hash)
+    }
+
+    pub fn get_prev_epoch_last_block_hash(&self, epoch_id: &String) -> Result<String, Box<dyn Error>> {
+        let current_epoch_first_block_hash = self.get_current_epoch_first_block_hash(epoch_id).unwrap();
+        let current_epoch_first_block_response: BlockResponse = self
+            .client
+            .post(&self.rpc_endpoint)
+            .json(&serde_json::json!({
+                "jsonrpc": "2.0",
+                "id": "dontcare",
+                "method": "block",
+                "params": {
+                    "block_id": current_epoch_first_block_hash
+                }
+            }))
+            .send()?
+            .json()?;
+        let last_epoch_last_block_hash = current_epoch_first_block_response.result.header.prev_hash;
+        Ok(last_epoch_last_block_hash)
+    }
+
+    pub fn get_prev_epoch_id(&self, epoch_id: &String) -> Result<String, Box<dyn Error>> {
+        let prev_epoch_last_block_hash = self.get_prev_epoch_last_block_hash(epoch_id).unwrap();
+        let prev_epoch_last_block_response: BlockResponse = self
+            .client
+            .post(&self.rpc_endpoint)
+            .json(&serde_json::json!({
+                "jsonrpc": "2.0",
+                "id": "dontcare",
+                "method": "block",
+                "params": {
+                    "block_id": prev_epoch_last_block_hash
+                }
+            }))
+            .send()?
+            .json()?;
+        Ok(prev_epoch_last_block_response.result.header.epoch_id)
+    }
+
+    // get the state root of previous epoch. This should be the state root of the last block of last epoch.
+    pub fn get_prev_epoch_state_root(&self, block_hash: &String) -> Result<String, Box<dyn Error>> {
+        let prev_epoch_last_block_hash = self.get_prev_epoch_last_block_hash(block_hash).unwrap();
+        let prev_epoch_last_block_response: BlockResponse = self
+            .client
+            .post(&self.rpc_endpoint)
+            .json(&serde_json::json!({
+                "jsonrpc": "2.0",
+                "id": "dontcare",
+                "method": "block",
+                "params": {
+                    "block_id": prev_epoch_last_block_hash
+                }
+            }))
+            .send()?
+            .json()?;
+        let chunks = prev_epoch_last_block_response.result.chunks;
+        let state_root = &chunks[chunks.len() -1].prev_state_root;
+        Ok(state_root.to_string())
+    }
+
+    pub fn get_latest_block_hash(&self) -> Result<String, Box<dyn Error>> { 
```

Comment:
As it's querying the final block and not the latest block, should this be named `get_final_block_hash()`?

```suggestion
    pub fn get_final_block_hash(&self) -> Result<String, Box<dyn Error>> { 
```